mmcv/transform/processing.py

```diff
@@ -0,0 +1,338 @@
+from .base import BaseTransform
+from mmcv.image.geometric import _scale_size
+
+
+@TRANSFORMS.register_module()
+class Normalize(BaseTransform):
+    """Normalize the image.
+    Required key is "img". Added or updated keys is "img_norm_cfg",which
+    contain 'mean', 'std', 'to_rgb'.
+
+    Args:
+        mean (sequence): Mean values of 3 channels.
+        std (sequence): Std values of 3 channels.
+        to_rgb (bool): Whether to convert the image from BGR to RGB,
+            default is true.
+    """
+
+    def __init__(self,
+                 mean: Sequence[float],
+                 std: Sequence[float],
+                 to_rgb: bool = True) -> None:
+        self.mean = np.array(mean, dtype=np.float32)
+        self.std = np.array(std, dtype=np.float32)
+        self.to_rgb = to_rgb
+
+    def transform(self, results: dict) -> dict:
+        """Function to normalize images.
+
+        Args:
+            results (dict): Result dict from loading pipeline.
+
+        Returns:
+            dict: Normalized results, 'img_norm_cfg' key is added into
+                result dict.
+        """
+
+        results['img'] = mmcv.imnormalize(results['img'], self.mean, self.std,
+                                          self.to_rgb)
+        results['img_norm_cfg'] = dict(
+            mean=self.mean, std=self.std, to_rgb=self.to_rgb)
+        return results
+
+    def __repr__(self) -> str:
+        repr_str = self.__class__.__name__
+        repr_str += f'(mean={self.mean}, std={self.std}, to_rgb={self.to_rgb})'
+        return repr_str
+
+
+@TRANSFORMS.register_module()
+class Resize(BaseTransform):
+    """Resize images & bbox & seg & keypoints.
+
+    This transform resizes the input image according to `scale` or
+    `scale_factor`. Bboxes, seg map and keypoints are then resized with the
+    same scale factor.
+
+    Args:
+        scale (int or tuple): Images scales for resizing. Default to None
+        scale_factor (float or tuple[float]): Scale factors for resizing.
+            Default to None.
+        keep_ratio (bool): Whether to keep the aspect ratio when resizing the
+            image.
+        bbox_clip_border (bool, optional): Whether to clip the objects outside
+            the border of the image. In some dataset like MOT17, the gt bboxes
+            are allowed to cross the border of images. Therefore, we don't
+            need to clip the gt bboxes in these cases. Defaults to True.
+        backend (str): Image resize backend, choices are 'cv2' and 'pillow'.
+            These two backends generates slightly different results. Default
+            to 'cv2'.
+        interpolation (str): Interpolation method, accepted values are
+            "nearest", "bilinear", "bicubic", "area", "lanczos" for 'cv2'
+            backend, "nearest", "bilinear" for 'pillow' backend. Default
+            to 'cv2'.
+    """
+
+    def __init__(self,
+                 scale: Optional[Union[int, Tuple[int]]] = None,
```

Comment:
I think there is no need to embed Union into Optional

Reply:
It is necessary because it allows None as input I think

Reply:
Sorry, I misunderstand this